### DISCUSSION TITLE
Database Migrations (Prisma?)

### DISCUSSION THREAD

**Discussion by jaredcosulich** [supabase/supabase#5870] [Questions]
I'm currently using Prisma for database migrations but am running into issues with the auth.uid() function. 

As I have RLS in the database I get this error when trying to run a migration:

```
Migration `20220308221326_init` failed to apply cleanly to the shadow database.
Error:
db error: ERROR: function auth.uid() does not exist
```

I need to create the auth.uid() function in the shadow database, but I don't want to overwrite it in the actual database. Is there any recommended way to do this? Or should I use something other than Prisma for database migrations?

Here is the relevant part of the initial migration:

```
CREATE SCHEMA IF NOT EXISTS "auth";
[How can I check for auth.uid() here?]

CREATE SCHEMA IF NOT EXISTS "extensions";
CREATE EXTENSION IF NOT EXISTS "uuid-ossp" SCHEMA extensions;
CREATE EXTENSION IF NOT EXISTS "moddatetime" SCHEMA extensions;
```

Any help would be greatly appreciated.

**Comment by NSikimic** [comment]
Hi @jaredcosulich , any luck in finding the solution? Im stuck on the same problem

**Comment by johannbuscail** [comment]
Hi @jaredcosulich , have you found the solution ?

**Comment by johnnyicon** [comment]
Has anyone resolved this? I'm having this issue as well.
